packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1470,6 +1471,44 @@ export class KeyringController extends BaseController<
     }
   }
 
+  /**
+   * Exports the vault encryption key.
+   *
+   * @returns The vault encryption key.
+   */
+  async exportEncryptionKey(): Promise<string> {
+    this.#assertIsUnlocked();
+
+    return await this.#withControllerLock(async () => {
+      // There is a case where the controller is unlocked but the encryption key
+      // is not set, even when #cacheEncryptionKey is true. This happens when
+      // calling changePassword with the existing password. In this case, the
+      // encryption key is deleted, but the state is not recreated, because the
+      // session state does not change in this case, and #updateVault is not
+      // called in #persistOrRollback.
+      if (!this.state.encryptionKey) {
+        return await this.#withVaultLock(async () => {
+          assertIsExportableKeyEncryptor(this.#encryptor);
+          assertIsValidPassword(this.#password);
+          const result = await this.#encryptor.decryptWithDetail(
+            this.#password,
+            // Ignoring undefined. Assuming vault is set when unlocked.
+            this.state.vault as string,
+          );
+          if (this.#cacheEncryptionKey) {
+            this.update((state) => {
+              state.encryptionKey = result.exportedKeyString;
+              state.encryptionSalt = result.salt;
+            });
+          }
```

Comment:
Curious: why are we updating this state here if we already used the salt from the vault, and the key is presumably the same?

Reply:
changed in https://github.com/MetaMask/core/commit/fee56528a060cb620a2fccc5ad1378f8a2afff45